src/averageAges.js

```diff
@@ -36,8 +43,23 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const allWomen = people.filter(person => person.sex === 'f');
+
+  const mothers = people.map(person => person.mother);
+
+  const womenWithChildren = allWomen.filter(
+    woman => mothers.includes(woman.name)
+  );
+
+  const women = withChildren ? womenWithChildren : allWomen;
+
+  const agesOfWomen = women.map(woman => woman.died - woman.born);
+
+  const sumOfAges = agesOfWomen.reduce((a, b) => (a + b), 0);
```

Comment:
seems you use this logic a couple of times - it seems reasonable to create a helper function for that

Reply:
Totally! It looks much better with a separate function. Thx:fire: